src/Compositor.cpp

```diff
@@ -2352,7 +2353,21 @@ void CCompositor::setWindowFullscreenState(const PHLWINDOW PWINDOW, sFullscreenS
     const eFullscreenMode CURRENT_EFFECTIVE_MODE = (eFullscreenMode)std::bit_floor((uint8_t)PWINDOW->m_sFullscreenState.internal);
     const eFullscreenMode EFFECTIVE_MODE         = (eFullscreenMode)std::bit_floor((uint8_t)state.internal);
 
-    const bool            CHANGEINTERNAL = !(PWINDOW->m_bPinned || CURRENT_EFFECTIVE_MODE == EFFECTIVE_MODE || (PWORKSPACE->m_bHasFullscreenWindow && !PWINDOW->isFullscreen()));
+	if (*PALLOWPINFULLSCREEN) {
```

Comment:
can't these 2 ifs for *PALLOWPINFULLSCREEN be merged?

Reply:
ah ya, i just try to not confuse my self when looking at the logic, fogot to put them back, thanks for reminding